Cython/Compiler/ExprNodes.py

```diff
@@ -6949,6 +6949,8 @@ def analyse_as_type_attribute(self, env):
                         ubcm_entry.func_cname = entry.func_cname
                         ubcm_entry.is_unbound_cmethod = 1
                         ubcm_entry.scope = entry.scope
+                        if type.is_cpp_class:
+                            ubcm_entry.overloaded_alternatives = [Symtab.Entry(entry.name, entry.func_cname, entry.type) for entry in entry.overloaded_alternatives]
```

Comment:
An `Entry` is actually never created like this, rather through some `declare_*()` method in a Scope (sub-)class. I'm not sure right now if this qualifies as a special case. Why do we need a copy here?

Reply:
Stating the obvious: I'm filling the `entry.overloaded_alternatives` array because polymorphism is allowed in a cppclass, and PyrexTypes.best_match must be able to choose among the alternatives.

Please notice the ubcm_entry creation code is `ubcm_entry = Symtab.Entry(entry.name, cname, ctype)` **but**, in the case of a cppclass, `cname = entry.func_cname` (with the exception of fused_types, a cfunction entry in CppClassScope always has a func_cname).

After setting things in ubcm_entry we will return it, wrapped in a NameNode which will uses `ubcm_entry.cname` during code generation.

`ubcm_entry.cname = entry.func_cname` is then required to have a full namespace path when generating the C++ code. This way we will have `Namespace::Class::Subclass::method` in the generated code. With `ubcm_entry.cname = entry.cname` we would just have `method` in the generated code, and the C++ compiler would be misleaded. This is the same problem for the alternatives in ubcm_entry.overloaded_alternatives.

We cannot call a declare_*() for the alternatives, because, like ubcm_entry, we are not creating a new entry, we are just copying it (with a cname change).